src/main/java/core/basesyntax/strategy/handler/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Transaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseOperationHandler implements OperationHandler {
+
+    @Override
+    public void apply(Transaction transaction) {
+        Fruit fruit = transaction.getFruit();
+        Integer currentQuantity = Storage.get(fruit);
+        Storage.put(fruit, (currentQuantity == null ? 0
+                : currentQuantity) - transaction.getQuantity());
```

Comment:
maybe let's throw an exception if there is no such fruit or we don't have enough of it?